OpenRA.Mods.Common/Traits/Render/WithAttackAnimation.cs

```diff
@@ -42,23 +45,35 @@ public class WithAttackAnimation : ITick, INotifyAttack
 		readonly WithAttackAnimationInfo info;
 		readonly AttackBase attack;
 		readonly Armament armament;
-		readonly WithSpriteBody wsb;
+		readonly WithSpriteBody[] wsbs;
+		readonly bool onlyAttackAnim;
 
 		int tick;
+		bool attackAnimPlaying;
 
 		public WithAttackAnimation(ActorInitializer init, WithAttackAnimationInfo info)
 		{
 			this.info = info;
 			attack = init.Self.Trait<AttackBase>();
 			armament = init.Self.TraitsImplementing<Armament>()
 				.Single(a => a.Info.Name == info.Armament);
-			wsb = init.Self.Trait<WithSpriteBody>();
+			wsbs = init.Self.TraitsImplementing<WithSpriteBody>().Where(w => info.BodyNames.Contains(w.Info.Name)).ToArray();
+			onlyAttackAnim = string.IsNullOrEmpty(info.AimSequence) && string.IsNullOrEmpty(info.ReloadPrefix);
 		}
 
 		void PlayAttackAnimation(Actor self)
 		{
 			if (!string.IsNullOrEmpty(info.AttackSequence))
-				wsb.PlayCustomAnimation(self, info.AttackSequence, () => wsb.CancelCustomAnimation(self));
```

Comment:
Wouldn't a `return` here also make Tick return early even if there's actually an aim- or reload sequence, or would that `return` actually be limited to this method and not bleed through?

Returning if there's no matching body wouldn't be a problem either way, because in that case the aim/reload wouldn't play, either.

That's the reasoning why I did it this way, in any case.